spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -199,4 +200,17 @@ protected RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		return rt;
 	}
 
+	/**
+	 * Apply {@link AutowireCapableBeanFactory#initializeBean(Object, String) initializeBean} to the
+	 * {@code target} with the empty string {@code ""} as the {@code beanName} parameter.
+	 * @param target the {@code Object} to initialize
+	 * @see AutowireCapableBeanFactory#initializeBean(Object, String)
+	 */
+	protected void applyInitializeBean(Object target) {
+		ApplicationContext applicationContext = getApplicationContext();
+		if (applicationContext != null) {
+			applicationContext.getAutowireCapableBeanFactory()
+				.initializeBean(target, "");
```

Comment:
Is the empty string for the `beanName` okay here?

Reply:
We probably need to think making some bean name over here since it is going to be propagated down to the `setBeanName()` on the target component.
I suppose we can reply on the provided destination name and group. And add some prefix or suffix to make it clear that it is consumer or producer.

Reply:
> We probably need to think making some bean name over here since it is going to be propagated down to the `setBeanName()` on the target component.
> I suppose we can reply on the provided destination name and group. And add some prefix or suffix to make it clear that it is consumer or producer.

I have taken a random guess, but I feel in no way qualified to decide on a naming scheme here.  Should we be using the class name, maybe?

I'm not sure what the consequences are of giving a name here which isn't in the BeanFactory... I can say from personal experience that it can be very confusing to see these objects appear to be Beans when debugging, but not actually be able to get them from the context.  I feel like the name should give some kind of clue where it came from, and that it _isn't_ a Bean, if possible 🤷